kernel/kernel-idgenerator-service/src/main/java/io/mosip/kernel/vidgenerator/service/impl/VidServiceImpl.java

```diff
@@ -122,14 +133,14 @@ public void expireAndRelease() {
 
 	private void expireEligibleVids() {
 		List<VidAssignedEntity> vidAssignedEntities = vidAssignedRepository
-			.findByStatusAndIsDeletedFalse(VidLifecycleStatus.ASSIGNED);
+			.findByStatusAndIsDeletedFalse(VidLifecycleStatus.ASSIGNED,vidExpireLimit);
```

Comment:
Since this is about expiry and there is no sorting, if we add limit will it not create issues like taking more time to expire when there are large no of rows ? What is the frequency of this job ?

Reply:
yes it will take some time to expire. frequency is per day at 11 PM. But they can increase freq and limit or should we remove limit as we will scan and expire all in case eligible?